library/src/main/kotlin/io/intrepid/bleidiom/BleIdiomService.kt

```diff
@@ -181,7 +184,7 @@ open class BleService<Svc : BleService<Svc>> : BleConfigureDSL<Svc> {
     fun observeConnectionState() = device.observeConnectionState()
 
     fun discoverPrimaryServices(timeout: Long = 20_000) =
-            sharedConnection.take(1).flatMapTry { it.discoverServices(timeout, TimeUnit.MILLISECONDS).toRx2() }
+            sharedConnection.take(1).flatMapTry { it.discoverServices(timeout, TimeUnit.MILLISECONDS).toObservable() }
```

Comment:
For some reason, RxAndroidBle folks think `discoverServices` emits a value only once. If this is indeed correct, the return method for `discoverPrimaryServices()` should reflect that. Returning a number (list of values) that completes. Calling `toObservable()` may still be the right thing to do here, though... not sure...

Reply:
duscoverServices may return multiple values depending on the services device supports. Eg. same device can support temperature monitor and heart rate monitor.
I think toObservable is fine.